source/Nabla/CIrrDeviceStub.h

```diff
@@ -99,8 +99,15 @@ namespace nbl
             //! Remove all messages pending in the system message loop
             virtual void clearSystemMessages();
 
+            EGLDisplay getEGLDisplay() const { return Display; }
+
         protected:
 
+            virtual bool switchToFullscreen(bool reset = false) = 0;
+
+            //! create the driver
+            void createDriver();

```

Comment:
we're getting rid of the `device` entirely (eventually) start abandoning the ship